include/robot_state_publisher/robot_state_publisher.hpp

```diff
@@ -51,56 +51,117 @@ using MimicMap = std::map<std::string, urdf::JointMimicSharedPtr>;
 namespace robot_state_publisher
 {
 
+/// A class that represents a mapping between a KDL segment and its root and tip.
 class SegmentPair final
 {
 public:
+  /// Constructor
   explicit SegmentPair(
     const KDL::Segment & p_segment,
     const std::string & p_root,
     const std::string & p_tip)
   : segment(p_segment), root(p_root), tip(p_tip) {}
 
-  KDL::Segment segment;
-  std::string root;
-  std::string tip;
+  KDL::Segment segment;  ///< The KDL segment
+  std::string root;  ///< The name of the root element to which this link is attached
+  std::string tip;  ///< The name of the element
 };
 
+/// The class that contains all of the functionality of the RobotStatePublisher ROS 2 node.
 class RobotStatePublisher : public rclcpp::Node
 {
 public:
   /// Constructor
   explicit RobotStatePublisher(const rclcpp::NodeOptions & options);
 
 protected:
-  /** Publish transforms to tf
-   * \param joint_positions A map of joint names and joint positions.
-   * \param time The time at which the joint positions were recorded
+  /** Setup the URDF for use.
+   * This method first parses the URDF into an internal representation.
+   * Based on that representation, it then generates the list of joint segments
+   * and mimic pairs that it needs during runtime.  Finally, it publishes
+   * the text of the URDF to the network on the /robot_description topic.
+   *
+   * \param[in] urdf_xml The string representing the URDF XML.
+   */
```

Comment:
I don't know what the prevailing style in this repository is, and if this is your preference that's fine too, but the style in `rclcpp` (and other core packages) would be this:

```suggestion
  /// Setup the URDF for use.
  /**
   * This method first parses the URDF into an internal representation.
   * Based on that representation, it then generates the list of joint segments
   * and mimic pairs that it needs during runtime.  Finally, it publishes
   * the text of the URDF to the network on the /robot_description topic.
   *
   * \param[in] urdf_xml The string representing the URDF XML.
   */
```